Automattic/pocket-casts-ios
# Performance: move DROP operations outside of migrations

The drop of columns on `SJEpisode` and `SJPodcast` it's a very heavy operation for big databases.

We moved them out of the migration so it happens when the app is fully launched to avoid a watchdog termination during startup.

A dialog is displayed while the operation is going on (which for huge databases takes some seconds).

<img src="https://github.com/Automattic/pocket-casts-ios/assets/7040243/2aed5d7c-8a10-4fba-8ca2-865a47e693e8" width="300">

## To test

1. Run a previous build with a large database (preferably with more than 1k podcasts)
2. Run this branch
3. ✅ You should see the alert after the app opens
4. ✅ After a while it should go away
5. Kill the app
6. Reopen it
7. ✅ You should not see any dialog

## Checklist
